src/accommodation-agent/Tools/AccommodationTools.cs

```diff
@@ -0,0 +1,133 @@
+using System.ComponentModel;
+using System.Text.Json;
+using Microsoft.Extensions.AI;
+using AccommodationAgent.Services;
+using AccommodationAgent.Models;
+
+namespace AccommodationAgent.Tools;
+
+public class AccommodationTools
+{
+    private readonly IAccommodationService _accommodationService;
+    private readonly RerankingService _rerankingService;
+    private readonly ILogger<AccommodationTools> _logger;
+
+    // Known landmarks with coordinates
+    private static readonly Dictionary<string, (double Latitude, double Longitude)> KnownLandmarks = new(StringComparer.OrdinalIgnoreCase)
+    {
+        { "colosseum", (41.8902, 12.4922) },
+        { "coliseum", (41.8902, 12.4922) },
+        { "roman forum", (41.8925, 12.4853) },
+        { "vatican", (41.9029, 12.4534) },
+        { "pantheon", (41.8986, 12.4768) },
+        { "trevi fountain", (41.9009, 12.4833) },
+        { "spanish steps", (41.9058, 12.4823) },
+        { "trastevere", (41.8899, 12.4707) }
+    };
+
+    public AccommodationTools(
+        IAccommodationService accommodationService,
+        RerankingService rerankingService,
+        ILogger<AccommodationTools> logger)
+    {
+        _accommodationService = accommodationService;
+        _rerankingService = rerankingService;
+        _logger = logger;
+    }
+
+    [Description("Search for accommodations based on various criteria including rating, location, amenities, price, and type")]
+    public async Task<string> SearchAccommodationsAsync(
+        [Description("The user's original search query to use for reranking results")] string userQuery,
+        [Description("Minimum user rating from 1 to 5 (e.g., 4 means at least 4 stars)")] double? minRating = null,
+        [Description("City name to search in (e.g., 'Rome', 'Latina')")] string? city = null,
+        [Description("Landmark or place name to search near (e.g., 'Colosseum', 'Vatican')")] string? nearLandmark = null,
```

Comment:
Don't use landmark names as reference point. Always use coordinates as tool inputs. Also define a tool to geocode addresses and landmarks. This tool can (for now) use mocked data but will be then replaced by a real geocoding api (Azure Maps API probably).